test/hotspot/jtreg/compiler/loopopts/TestCMoveLimitType.java

```diff
@@ -0,0 +1,138 @@
+/*
+ * Copyright (c) 2023, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+/*
+ * @test
+ * @bug 8299975
+ * @summary Limit underflow protection CMoveINode in PhaseIdealLoop::do_unroll must also protect type from underflow
+ * @run main/othervm -XX:+UnlockDiagnosticVMOptions -XX:-TieredCompilation
+ *                   -XX:CompileCommand=compileonly,compiler.loopopts.TestCMoveLimitType::test*
+ *                   -XX:CompileCommand=dontinline,compiler.loopopts.TestCMoveLimitType::dontInline
+ *                   -XX:RepeatCompilation=50 -XX:+StressIGVN
```

Comment:
You could also remove `RepeatCompilation` here and in the run below. If there is a problem at some point, we would eventually catch it by re-executing the test over and over again in our testing due to having a different seed each time.

Reply:
Maybe you can also add `@key stress randomness`.

Reply:
Ok, added the keywords, but will leave the `RepeatCompilation`.